src/lib/server/websearch/markdown/utils/stringify.ts

```diff
@@ -0,0 +1,72 @@
+import type { SerializedHTMLElement } from "../../scrape/types";
+import { MarkdownElementType, type MarkdownElement } from "../types";
+
+// --- Markdown Elements ---
+
+/** Converts markdown element to a string with formatting */
+export function stringifyMarkdownElement(elem: MarkdownElement): string {
+	const content = elem.content.trim();
+	if (elem.type === MarkdownElementType.Header) return `${"#".repeat(elem.level)} ${content}\n\n`;
+	if (elem.type === MarkdownElementType.BlockQuote) {
+		return `${"> ".repeat(elem.depth)}${content}\n\n`;
+	}
+	if (elem.type === MarkdownElementType.CodeBlock) return `\`\`\`\n${content}\n\`\`\`\n\n`;
```

Comment:
```suggestion
	if (elem.type === MarkdownElementType.CodeBlock) return `\`\`\`\${elem.codeLanguage}n${content}\n\`\`\`\n\n`;
```

nit, it is possible to get the code language information like this?

Reply:
eventually yes! but for now, there's no easy way to detect it that I know of. It'll likely involving looking at the class names on the pre/code elems